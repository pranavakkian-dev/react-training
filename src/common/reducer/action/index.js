import {
  LOGIN, RESET,
} from '../../constants/reducerTypes';

export const handleLogin = (username) => ({
  type: LOGIN,
  payload: username,
});

export const handleReset = () => ({
  type: RESET,
});
