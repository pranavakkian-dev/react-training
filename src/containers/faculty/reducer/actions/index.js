import { FACULTY_RESET, ADD_FACULTY, DELETE_FACULTY } from '../../../../common/constants/reducerTypes';

export const handleAddFaculty = (faculty) => ({
  type: ADD_FACULTY,
  payload: faculty,
});

export const handleDeleteFaculty = (faculty) => ({
  type: DELETE_FACULTY,
  payload: faculty,
});

export const handleFacultyReset = () => ({
  type: FACULTY_RESET,
});
