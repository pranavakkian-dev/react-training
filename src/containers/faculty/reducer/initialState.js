const initialState = {
  faculties: [],
};

export default initialState;
