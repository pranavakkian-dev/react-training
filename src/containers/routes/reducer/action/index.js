import {
  LOGIN, RESET,
} from '../../../../common/constants/reducerTypes';

export const handleLogin = (username) => ({
  type: LOGIN,
  payload: username,
});

export const handleReset = () => ({
  type: RESET,
});
