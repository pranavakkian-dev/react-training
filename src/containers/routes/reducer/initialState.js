const initialState = {
  loggedIn: false,
  userName: null,
};

export default initialState;
