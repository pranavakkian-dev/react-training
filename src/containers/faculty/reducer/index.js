import initialState from './initialState';
import { FACULTY_RESET, ADD_FACULTY, DELETE_FACULTY } from '../../../common/constants/reducerTypes';

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case ADD_FACULTY: {
      const { faculties } = state;
      return { ...state, faculties: [...faculties, { ...payload }] };
    }
    case DELETE_FACULTY: {
      const { faculties } = state;
      const updatedFaculties = faculties.filter((faculty) => faculty !== payload);
      return { ...state, faculties: [...updatedFaculties] };
    }
    case FACULTY_RESET:
      return initialState;
    default:
      return state;
  }
};

export default reducer;
