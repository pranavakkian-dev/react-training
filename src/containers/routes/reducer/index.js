import initialState from './initialState';
import {
  LOGIN, RESET,
} from '../../../common/constants/reducerTypes';

const reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case LOGIN:
      return { ...state, loggedIn: true, userName: payload };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

export default reducer;
