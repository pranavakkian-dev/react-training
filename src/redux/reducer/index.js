import initialState from './initialState';
import {
  LOGIN, RESET,
} from '../../common/constants/reducerTypes';

const Reducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case LOGIN:
      return { ...state, loggedIn: true, userName: payload };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

export default Reducer;
