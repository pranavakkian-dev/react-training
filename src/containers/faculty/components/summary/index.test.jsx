import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Summary from './index';
import { handleDeleteFaculty } from '../../reducer/actions';

describe('Summary', () => {
  const faculties = [
    {
      id: '1',
      name: 'user1',
    },
    {
      id: '2',
      name: 'user2',
    },
  ];
  const mockedDispatch = jest.fn();

  test('Component rendered successfully!', () => {
    const { container } = render(<Summary faculties={faculties} dispatch={mockedDispatch} />);
    expect(container).toMatchSnapshot();
  });

  test('Table rendered successfully!', () => {
    render(<Summary faculties={faculties} dispatch={mockedDispatch} />);
    expect(screen.getByRole('table')).toBeInTheDocument();
  });

  test('Table header rendered successfully!', () => {
    render(<Summary faculties={faculties} dispatch={mockedDispatch} />);
    expect(screen.getByRole('columnheader', { name: 'ID' })).toBeInTheDocument();
    expect(screen.getByRole('columnheader', { name: 'Name' })).toBeInTheDocument();
  });

  test('Table data rendered successfully!', () => {
    render(<Summary faculties={faculties} dispatch={mockedDispatch} />);
    expect(screen.getByRole('cell', { name: '1' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: 'user1' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: '2' })).toBeInTheDocument();
    expect(screen.getByRole('cell', { name: 'user2' })).toBeInTheDocument();
  });

  test('Delete faculty handled successfully!', () => {
    render(<Summary faculties={faculties} dispatch={mockedDispatch} />);
    userEvent.click(screen.getAllByRole('button', { name: 'Delete' })[0]);
    expect(mockedDispatch).toHaveBeenNthCalledWith(1, handleDeleteFaculty({ id: '1', name: 'user1' }));
  });
});
