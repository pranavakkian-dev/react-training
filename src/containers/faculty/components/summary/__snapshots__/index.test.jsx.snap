// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Component rendered successfully! 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          user1
        </td>
        <td>
          <button
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          user2
        </td>
        <td>
          <button
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
